server/src/test/java/org/elasticsearch/action/admin/indices/alias/GetAliasesResponseTests.java

```diff
@@ -0,0 +1,249 @@
+    protected GetAliasesResponse mutateInstance(GetAliasesResponse response) {
+        switch (randomInt(2)) {
+        case 0:
+            return new GetAliasesResponse(mutateAliases(response.getAliases()), response.status(), response.errorMsg());
+        case 1:
+            return new GetAliasesResponse(response.getAliases(),
+                    randomValueOtherThan(response.status(), () -> randomFrom(RestStatus.values())), response.errorMsg());
+        case 2:
+            if (response.status() == RestStatus.OK) {
+                return new GetAliasesResponse(response.getAliases(),
+                        randomValueOtherThan(response.status(), () -> randomFrom(RestStatus.values())), randomAlphaOfLengthBetween(0, 100));
+            }
+            return new GetAliasesResponse(response.getAliases(), response.status(), randomAlphaOfLengthBetween(5, 100));
+        default:
+            assert false;
+            return null;
+        }
+    }
+
+    private static ImmutableOpenMap<String, List<AliasMetaData>> mutateAliases(ImmutableOpenMap<String, List<AliasMetaData>> aliases) {
+        if (aliases.isEmpty()) {
+            return createIndicesAliasesMap(1, 3).build();
+        }
+
+        if (randomBoolean()) {
+            ImmutableOpenMap.Builder<String, List<AliasMetaData>> builder = ImmutableOpenMap.builder(aliases);
+            ImmutableOpenMap<String, List<AliasMetaData>> list = createIndicesAliasesMap(1, 2).build();
+            list.forEach(e -> builder.put(e.key, e.value));
+            return builder.build();
+        }
+
+        Set<String> indices = new HashSet<>();
+        Iterator<String> keys = aliases.keysIt();
+        while (keys.hasNext()) {
+            indices.add(keys.next());
+        }
+
+        List<String> indicesToBeModified = randomSubsetOf(randomIntBetween(1, indices.size()), indices);
+        ImmutableOpenMap.Builder<String, List<AliasMetaData>> builder = ImmutableOpenMap.builder();
+
+        for (String index : indices) {
+            List<AliasMetaData> list = new ArrayList<>(aliases.get(index));
+            if (indicesToBeModified.contains(index)) {
+                if (randomBoolean() || list.isEmpty()) {
+                    list.add(createAliasMetaData());
+                } else {
+                    int aliasIndex = randomInt(list.size() - 1);
+                    AliasMetaData aliasMetaData = list.get(aliasIndex);
+                    list.add(aliasIndex, mutateAliasMetaData(aliasMetaData));
+                }
+            }
+            builder.put(index, list);
+        }
+        return builder.build();
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        return p -> p.equals("") // do not add elements at the top-level as any element at this level is parsed as a new index
+                || p.endsWith(".aliases") // do not add new alias
+                || p.contains(".aliases."); // do not be testing the AlilasMetaData.fromContent
```

Comment:
I prefer not to test the `AliasMetaData#toXContent` as part of the `GetAliasesResponseTests`. Maybe it would be better to add such tests to `AliasMetaDataTests` ?

Reply:
I am not following. If you test `GetAliasesResponse`, you end up testing everything that it holds as well? It may be a good idea to test some parts separately, but I think that should not affect the exclusions here?